### PR TITLE
[Dotnet Monitor Experimental] Add getting started for metrics usage using dotnet-monitor

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
This PR introduces documentation that illustrates how to collect metrics using dotnet-monitor and leverage a Prometheus server for scraping and visualizing the data in Grafana.

**Motivation**: The primary goal of this documentation is to enable OpenTelemetry auto-instrumentation contributors to try out dotnet-monitor to identify the gaps. By providing a clear example and pathway, aim to facilitate a better understanding end to end flow.

The approach is based on a previous commit in the [OpenTelemetry .NET SDK documentation](https://github.com/open-telemetry/opentelemetry-dotnet/blob/874285194ba2a611506c4abff9a28247b94d56d0/docs/metrics/getting-started-prometheus-grafana/README.md)

**Readme preview** - https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/cafe8111cffea0052bae7865833cfc09c37f1efb/NextGen.AutoInstrumentation/docs/getting-started-dotnet-monitor-metrics/README.md

### CODE REVIEW

**Comment by pjanotti** [comment]
Thanks for getting started on this @rajkumar-rangaraj!

One question immediately pops to my head: if we add instrumentation that adds metrics but set the metric exporter to none, can we already use `dotnet-monitor` to capture the metrics? It sounds like it should work, but, I didn't try it yet.

**Comment by rajkumar-rangaraj** [comment]
> One question immediately pops to my head: if we add instrumentation that adds metrics but set the metric exporter to none, can we already use `dotnet-monitor` to capture the metrics? It sounds like it should work, but, I didn't try it yet.

It will work. In the current sample, I did not add the OpenTelemetry .NET SDK or any MetricListener. Nevertheless, we are still able to obtain the metrics.

File: OpenTelemetry.AutoInstrumentation.sln [hunk 1]
```diff
@@ -221,6 +221,10 @@ Project("{778DAE3C-4631-46EA-AA77-85C1314464D9}") = "TestApplication.ContinuousP
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "TestApplication.ContinuousProfiler.ContextTracking", "test\test-applications\integrations\TestApplication.ContinuousProfiler.ContextTracking\TestApplication.ContinuousProfiler.ContextTracking.csproj", "{B74CB036-10F5-449D-8CBA-44F77E68D042}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "NextGen.AutoInstrumentation", "NextGen.AutoInstrumentation", "{3F051815-8E0D-4356-BC36-55CA642DDF18}"
```
[the remark this answers is not in the document]
Comment (rajkumar-rangaraj) [reply OpenTelemetry.AutoInstrumentation.sln:224]: This is not a part of CI nor does it impact our binaries in any way. The idea is to ramp up faster and identify the gaps. We could maintain another branch, but that would be a maintenance overhead. As we plan to align ourselves, I believe this is a better approach.

[the remark this answers is not in the document]
Comment (birojnayak) [reply OpenTelemetry.AutoInstrumentation.sln:224]: if others are ok, I am fine.. generally we follow this as sample (if it's cooking and no relevance with OTEl yet).. let's see what others think and we can resolve based on that...

File: NextGen.AutoInstrumentation/Directory.Build.props
Comment (Kielek) [inline NextGen.AutoInstrumentation/Directory.Build.props:1]: Minor: Please rename folder `NextGen.AutoInstrumentation` to just `next-gen`. This is convention we have for all folders.

**Review by Kielek — APPROVED** [review]
LGTM. Thanks for detailed explanation how to setup dotnet-monitor.
Only one, minor comment.

File: OpenTelemetry.AutoInstrumentation.sln (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kielek) [reply OpenTelemetry.AutoInstrumentation.sln:224]: Keeping it in solution is not a problem for me.
Even more, I am not against including it in pipeline. We have some issue here and in contrib with samples which are not covered by CI.

**Review by nrcventura — APPROVED** [review]
This looks good and is easy to follow. I am also fine with this being part of the CI so that we can try to keep it up to date.

**Review by pjanotti — APPROVED** [review]

**Review by birojnayak — APPROVED** [review]
LGTM

**Comment by rajkumar-rangaraj** [comment]
@Kielek There is a timeout error for one of the links in validate-document. Do you know how to fix it?

**Pull request merged**